py/desisim/templates.py

```diff
@@ -753,9 +760,9 @@ def make_galaxy_templates(self, nmodel=100, zrange=(0.6, 1.6), magrange=(21.0, 2
                         format(np.sum(success == 0)))
 
         if restframe:
-            return outflux, self.basewave, meta
+            return outflux * 1e17 * u.erg / (u.cm**2 * u.s * u.Angstrom), self.basewave * u.Angstrom, meta
         else:
-            return outflux, self.wave, meta
+            return outflux * 1e17 * u.erg / (u.cm**2 * u.s * u.Angstrom), self.wave * u.Angstrom, meta
```

Comment:
Units bug. This returns an astropy Quantity object with true units of 1e-17 erg/(s cm2 Angstrom) but reported units without the 1e-17.

My preference would be to not use astropy units at all and just return `outflux * 1e17` and put in the docstring that the units are `1e-17 erg/(s cm2 Angstrom)`, though I recognize reasonable arguments for why to provide a units-ful object, especially at certain critical interfaces like this.  You also include units in the accompanying metadata table, which is good, which could be an additional argument in favor of using units here.

If you do want to include astropy units to return a Quantity object instead of a vanilla numpy array (your choice), then you shouldn't include the 1e17 factor here.  

i.e. don't try to force astropy to return something where `str(flux.unit).startswith('1e-17')`; astropy will always strip the prefactor for the units and apply it to the numbers to simplify the units.  e.g. this would have worked while carrying around unnecessary cancelations:
```python
scaledfluxunits = 1e-17 * u.erg / (u.cm**2 * u.s * u.Angstrom)
return (outflux*1e17) * scaledfluxunits
```
but even then, flux.unit would be `erg / (Angstrom cm2 s)` and the values would be O(1e-17)...

Same comment applies to other units-ful return values in other `make_templates` functions.